src/security.ts

```diff
@@ -167,3 +170,60 @@ export const iamMfaEnabledForConsoleAccess: ResourceValidationPolicy = {
     }),
 };
 registerPolicy("iamMfaEnabledForConsoleAccess", iamMfaEnabledForConsoleAccess);
+
+
+/** @internal */
+// Enforce the note on aws.iam.Role:
+//
+// NOTE: If you use this resource’s managed_policy_arns argument or inline_policy configuration blocks, this resource
+// will take over exclusive management of the role's respective policy types (e.g., both policy types if both arguments
+// are used). These arguments are incompatible with other ways of managing a role's policies, such as
+// aws.iam.PolicyAttachment, aws.iam.RolePolicyAttachment, and aws.iam.RolePolicy. If you attempt to manage a role’s
+// policies by multiple means, you will get resource cycling and/or errors.
+export const iamRoleNoPolicyManagementConflicts: StackValidationPolicy = {
+    name: "iam-role-no-policy-management-conflicts",
+    description: "Checks that iam.Role resources do not conflict with iam.PolicyAttachment, iam.RolePolicyAttachment, iam.RolePolicy",
+    validateStack: (args: StackValidationArgs, reportViolation: ReportViolation) => {
+        args.resources.forEach(r => {
+            var roleProp: string;
+            var currentType: string;
+            switch(r.type) {
+                case "aws:iam/policyAttachment:PolicyAttachment": {
+                    roleProp = "roles";
+                    currentType = "PolicyAttachment";
+                    break
+                }
+                case "aws:iam/rolePolicyAttachment:RolePolicyAttachment": {
+                    roleProp = "role";
+                    currentType = "RolePolicyAttachment";
+                    break
+                }
+                case "aws:iam/rolePolicy:RolePolicy": {
+                    roleProp = "role";
+                    currentType = "RolePolicy";
+                    break
+                }
+                default: {
+                    return
+                }
+            }
+
+            if (r.propertyDependencies[roleProp]) {
+                r.propertyDependencies[roleProp].forEach(dep => {
+                    if (dep.type !== "aws:iam/role:Role" || !dep.props) {
+                        return
+                    }
+                    if (dep.props["managedPolicyArns"] && dep.props["managedPolicyArns"].length > 0) {
+                        reportViolation(`${currentType} should not be used with a role ${dep.urn} that defines managedPolicyArns`, r.urn);
+                    }
+                    if (dep.props["inlinePolicies"] && dep.props["inlinePolicies"].length > 0) {
+                        reportViolation(`${currentType} should not be used with a role ${dep.urn} that defines inlinePolicies ${JSON.stringify(dep.props["inlinePolicies"])}`, r.urn);
```

Comment:
```suggestion
                        reportViolation(`${currentType} ${r.name} should not be used with a role ${dep.urn} that defines inlinePolicies ${JSON.stringify(dep.props["inlinePolicies"])}`, r.urn);
```

Should we also include the name of the resource in the message to make it easier to find?

Reply:
Since the message accepts r.urn which already contains the name, the display positions it specifically as pertaining to that resource I think, I think it's OK or we can fix the display to do a better job.